src/creature.cpp

```diff
@@ -685,7 +685,8 @@ void Creature::deal_projectile_attack( Creature *source, dealt_projectile_attack
         if( z ) {
             if( !proj.get_drop().is_null() ) {
                 z->add_effect( effect_tied, 1_turns, num_bp, true );
-                z->tied_item = proj.get_drop();
+                item drop_item = proj.get_drop();
+                z->tied_item.reset( &drop_item );
```

Comment:
`reset` transfers ownership of the pointer into the `value_ptr` object. It will delete the item later. Using a pointer to a function local object will cause UB, as it 's deleted when it goes out of scope *and* again when the `value_ptr` deletes it.

```suggestion
                z->tied_item = cata::make_value<item>( proj.get_drop() );
```
Applies to most of the other calls to `reset`.

(`make_value` is basically the same as `std::make_unique`.)